olive/evaluator/metric_config.py

```diff
@@ -90,3 +90,26 @@ def check_value(cls, v, values):
         if values["type"] in ["min-improvement", "max-degradation"] and v < 0:
             raise ValueError(f"Value must be positive for type {values['type']}")
         return v
+
+
+class SubTypeMetricResult(ConfigBase):
+    value: Union[float, int]
+    priority_rank: int
+    higher_is_better: bool
+
+
+class MetricResult(ConfigDictBase):
+    __root__: Dict[str, SubTypeMetricResult] = None
+
+
+class SignalResult(ConfigDictBase):
+    __root__: Dict[str, MetricResult] = None
```

Comment:
I think instead of nesting the final results with `{"type name": {"subtype name": SubTypeMetricResult}}`, it might be better to return a flattened dict with the type and subtype keys flattened into a single identifier. 

Otherwise, the evaluation results being nested makes the evaluation signals unnecessarily complicated. The search results, algorithms, footprints need to parse the results with multiple loops. 

Once they are evaluated, I don't think the hierarchy of the subtypes matter?

Reply:
Updated.